### DISCUSSION TITLE
Example of async validation using Zod in Next 13

### DISCUSSION THREAD

**Discussion by joselvelez** [vazco/uniforms#1283] [General]
Hello,

I am looking at using uniforms to replace RHF. I've run into trouble with getting async validation working with RHF and am hoping that uniforms can make it easier. I've looked at the uniforms docs, but I cannot see an example that would help me see if it will be any easier with uniforms.

I am using Next 13 (app router), prisma ORM, with data stored in MongoDB. I have a form field with an email address. I need to do async validation before the form is submitted to ensure that the email is not already being used. 

Any help would be greatly appreciated!

**Comment by kestarumper** [comment]
Hi @joselvelez,
Thanks for bringing up this topic. After reading your question and then looking at [our section about asynchronous validation](https://uniforms.tools/docs/api-forms/#asynchronous-validation), we noticed that the docs are not specific enough. I've created a PR https://github.com/vazco/uniforms/pull/1285 that makes this more explicit.

For now, have a look at the modified contents in the PR and let us know if that's enough. Feel free to ask more questions if needed.
